### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by johanlantz**
Extra fix for real device detection on Android.

Includes #24, #25, #43, #44, #45

### CODE REVIEW

File: lib/utils/utilities.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ var kill    = require('tree-kill');
 
 var HEADING_LINE_PATTERN = /List of devices/m;
 var DEVICE_ROW_PATTERN   = /(emulator|device|host)/m;
+var DEVICE_ONLY_ROW_PATTERN   = /(^(?!.*(emulator)).*device.*$)/m;
```
Comment (janpio) [inline lib/utils/utilities.js]: this is the actual change for this PR - but it includes much more

**Comment by janpio** [comment]
I merged this PR into a new `johanlantz` branch here in the repo. This will it make easiest for me to separate the individual changes into individual PRs that I can merge, and the commits will still be attributed to you.

I closed all the other individual PRs.

**Comment by johanlantz** [comment]
👍

**Comment by janpio** [comment]
Now at: https://github.com/apache/cordova-paramedic/pull/56 https://github.com/apache/cordova-paramedic/pull/57 https://github.com/apache/cordova-paramedic/pull/58 https://github.com/apache/cordova-paramedic/pull/59

**Pull request merged**